pandas/core/groupby/groupby.py

```diff
@@ -821,6 +821,18 @@ def __iter__(self) -> Iterator[tuple[Hashable, NDFrameT]]:
         Generator yielding sequence of (name, subsetted object)
         for each group
         """
+        result = self.grouper.get_iterator(self._selected_obj, axis=self.axis)
+        if result.gi_frame.f_locals["self"].raise_warning_single_grouper:
```

Comment:
Instead of creating and passing around `raise_warning_single_grouper`, can you use `self.keys` here?

Reply:
it raises a mypy error 
`pandas/core/groupby/groupby.py:824: error: "BaseGroupBy[NDFrameT]" has no attribute "keys"  [attr-defined]`

Reply:
I see; you can add a type-hint similar to `group_keys : bool | lib.NoDefault` on this class. BaseGroupBy is only used by GroupBy and Resampler, both of which have a keys attribute. Can type-hint the same way they are in `GroupBy.__init__`.